### PR TITLE
add unittest for path_not_unique

### LINKED ISSUE #7446: Add test for `mlflow.utils.validation.path_not_unique`
### Willingness to contribute

Yes. I would be willing to contribute this feature with guidance from the MLflow community.

### Proposal Summary

I would like to add an unit test for the `path_not_unique` function bellow because there is no test in https://github.com/mlflow/mlflow/blob/master/tests/utils/test_validation.py. https://github.com/mlflow/mlflow/blob/1bc83df65f57e03335d2c4d30bf94bc1a6d31a9d/mlflow/utils/validation.py#L82-L84

I want to discuss how to test this function here.

### Motivation

> #### What is the use case for this feature?

> #### Why is this use case valuable to support for MLflow users in general?

> #### Why is this use case valuable to support for your project(s) or organization?

> #### Why is it currently difficult to achieve this use case?

### Details

_No response_

### What component(s) does this bug affect?

- [X] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### PR DESCRIPTION
**Pull request by tsugumi-sys**
Signed-off-by: tsugumi-sys <tidemark0105@gmail.com>

## Related Issues/PRs

Resolve #7446 

## What changes are proposed in this pull request?

(Please fill in changes proposed in this fix)

- Add unittests for `path_not_unique` function.

## How is this patch tested?

- [ ] Existing unit/integration tests
- [x] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

@WeichenXu123

### CODE REVIEW

**Comment by harupy** [comment]
@tsugumi-sys Thanks for the PR. We might not need `path_not_unique`. 

https://github.com/mlflow/mlflow/blob/d6625d6df85b61f3661ba90efc36f94511a5c23f/mlflow/utils/validation.py#L13-L15

We validate param and metric names using `_VALID_PARAM_AND_METRIC_NAMES` first. As you can see, we don't allow `/`.

**Comment by tsugumi-sys** [comment]
@harupy 
I see, how about validating artifact path?
- validate artifact path https://github.com/mlflow/mlflow/blob/f5d38bd934cf63c16cf10d8ff5bcebcd3b72aec3/mlflow/store/artifact/artifact_repo.py#L302-L306

**Comment by harupy** [comment]
Ah, I missed that one.

File: tests/utils/test_validation.py [hunk 1]
```diff
@@ -43,6 +46,13 @@
 ]
 
 
+def test_path_not_unique():
+    for unique_path in UNIQUE_PATHS:
+        assert not path_not_unique(unique_path)
+    for not_unique_path in NOT_UNIQUE_PATHS:
+        assert path_not_unique(not_unique_path)
```
Comment (harupy) [inline tests/utils/test_validation.py:53]: You can use `pytest.mark.parametrize` here.

```python

@pytest.mark.parametrize(
    ("path", "expected"),
    [
        ("a", True),
        ...
    ]
)
def test_path_not_unique(path, expected):
    assert path_not_unique(path) is expected
```

Comment (tsugumi-sys) [reply tests/utils/test_validation.py]: I applied your FB! https://github.com/mlflow/mlflow/pull/7447/commits/869bc324b3bc2e3fbee00bf937e7e727803bb41a

Comment (tsugumi-sys) [reply tests/utils/test_validation.py]: how do I re run the failed job https://github.com/mlflow/mlflow/actions/runs/3616870043/jobs/6095235587 ?

Comment (harupy) [reply tests/utils/test_validation.py]: Only maintainers can rerun jobs. I reran it.

**Review by harupy — APPROVED** [review]

**Comment by tsugumi-sys** [comment]
I think I can do some easy refactoring using `pytest.mark.parametrize` in `tests/utils/test_validation.py`. Is it okay to make a new PR for this?

**Comment by harupy** [comment]
@tsugumi-sys Sounds good!

**Pull request merged**